providers/implementations/encode_decode/decode_der2key.c

```diff
@@ -544,15 +544,32 @@ static void *rsa_d2i_PKCS8(void **key, const unsigned char **der, long der_len,
 
 static int rsa_check(void *key, struct der2key_ctx_st *ctx)
 {
+    int	valid;
+    int n, e_ok, d_ok, p_ok, q_ok, dmp1_ok, dmq1_ok, iqmp_ok;
+
     switch (RSA_test_flags(key, RSA_FLAG_TYPE_MASK)) {
     case RSA_FLAG_TYPE_RSA:
-        return ctx->desc->evp_type == EVP_PKEY_RSA;
+        valid = (ctx->desc->evp_type == EVP_PKEY_RSA);
+        break;
     case RSA_FLAG_TYPE_RSASSAPSS:
-        return ctx->desc->evp_type == EVP_PKEY_RSA_PSS;
+        valid = (ctx->desc->evp_type == EVP_PKEY_RSA_PSS);
+        break;
+    default:
+        /* Currently unsupported RSA key type */
+        valid = 0;
     }
 
-    /* Currently unsupported RSA key type */
-    return 0;
+    n = BN_num_bits(RSA_get0_n(key));
+    e_ok = (BN_num_bits(RSA_get0_e(key)) < n);
+    d_ok = (BN_num_bits(RSA_get0_d(key)) < n);
```

Comment:
Maybe I'm remembering this wrong, but I thought that *d* could have the same number of bits as *n*, i.e. that the operator should be `<=` here...